modules/iam_users/main.tf

```diff
@@ -0,0 +1,109 @@
+# IAM User via assume role for EKS related service access
+locals {
+  users = [
+    for u in var.iam_users : {
+      name  = u[0]
+      roles = u[1]
+    }
+  ]
+
+  roles = [
+    for r in var.iam_roles : {
+      name   = r[0]
+      policy = r[1]
+    }
+  ]
+
+  role_users = [
+    for r in local.roles : {
+      name   = r.name
+      policy = r.policy
+      users = [
+        for u in local.users :
+        u.name if contains(u.roles, r.name)
+      ]
+    }
+  ]
+
+  role_user_arns = [
+    for r in local.role_users : {
+      name = r.name
+      user_arns = [
+        for u in aws_iam_user.access_user :
+        u.arn if contains(r.users, trimprefix(u.name, "${var.project}-"))
+      ]
+    }
+  ]
+}
+
+# Create users
+resource "aws_iam_user" "access_user" {
+  count = length(local.users)
+  name  = "${var.project}-${local.users[count.index].name}"
+
+  tags = {
+    roles = join("/", local.users[count.index].roles)
+  }
+}
+
+# Create roles for both AWS and Kubernetes aws-iam-authenticator
+resource "aws_iam_role" "access_assumerole" {
+  count              = length(local.role_users)
+  name               = "${var.project}-kubernetes-${local.role_users[count.index].name}-${var.environment}"
+  assume_role_policy = data.aws_iam_policy_document.access_assumerole_policy[count.index].json
```

Comment:
Sorry, these shouldn't be roles. We'll either want groups or to attach policies directly to users. Probably groups.

Reply:
Well this is kind of combining two things.. We need a way to specify a user's access to AWS resources, and then also a way to specify access to k8s. AWS access should probably be controlled via groups, and k8s access should be via roles, but then we also need a way to specify the access within k8s that each role has. Right now it's hard-coded below.

Reply:
Maybe we can chat more about this tomorrow.

Reply:
done